javascript/ql/src/semmle/javascript/ApiGraphs.qll

```diff
@@ -35,6 +38,28 @@ module API {
       )
     }
 
+    /**
+     * Gets a reference to the API component represented by this node.
+     *
+     * For example, `require('fs').readFileSync` is a reference to the `readFileSync` member from the
+     * `fs` module.
+     *
+     * No local data-flow or type-tracking happens on the result, which means that in
+     * `const x = fs.readFile` only `fs.readFile` is a reference to the `readFile` member of `fs`,
+     * neither `x` nor any node that `x` flows to is a reference to this API component.
+     */
+    DataFlow::SourceNode getAReference() { Impl::use(this, result) }
+
+    /**
+     * Gets a call to the function represented by this API component.
+     */
+    DataFlow::CallNode getACall() { result = getReturn().getAReference() }
+
+    /**
+     * Gets an instantiation of the function represented by this API component.
+     */
+    DataFlow::NewNode getAnInstantiation() { result = getInstance().getAReference() }
```

Comment:
Because of how `ApiGraph`s are currently implemented, such a predicate could be implemented as: 

```CodeQL
DataFlow::InvokeNode getAnInvocation() { result = getReturn().getAReference() }
```

But I opted to instead do a union of `getACall()` and `getAnInstantiation()`, in order to keep the predicate more readable.